### PR TITLE
Add documentation on COMDATs, add COMDAT sections

### PR DESCRIPTION
**Pull request by dschuff**

### CODE REVIEW

**Comment by dschuff** [comment]
@sbc100 questions here:
I assume a live reference to any of the elements in the group prevents all of the elements from being GCed? (Across all objects? e.g. a live function from one object will keep a section from another object?)
Is there any interesting thing to document about naming, e.g. data segment names?

**Comment by sbc100** [comment]
Its not so much that the comdat element are either all live all all not live, its more that the entire group always comes from the same object file IIUC.    i.e. a a given object ends up being the defining object for a given group, all elements on that comdat in other object are ignored.  But some elements within the comdat group could still be exlucded if they are no needed.   I think of it more a weak group...  once one object defines a given group no other object gets a look in. 
  But I could be wrong in my thinking..

**Comment by dschuff** [comment]
Updated the sentence. Does that look right?

File: Linking.md [hunk 1]
```diff
@@ -509,7 +509,8 @@ COMDATs
 -------
 
 A COMDAT group may contain one or more functions, data segments, and/or custom sections.
-The linker will include only one of each of these elements with a given group name.
+The linker will include only one of each of these elements with a given group name,
```
Comment (sbc100) [inline Linking.md:512]: This still doesn't sounds quite right to me... can't there be N different function in a given group.. in which case the linker will be pick all N (from the same object file).

Or is it not possible to put many functions in a single comdat group in a single object file?

Comment (sbc100) [reply Linking.md:512]: My understanding would be more like:  "the linker will include all these elements from a single object file, and will ignore any elements in that from from other object files"?

Comment (dschuff) [reply Linking.md:512]: I'm not sure, that's why I asked :)

Comment (sbc100) [reply Linking.md:512]: I think the comdat group can contain any number of element of a given type.  So you might have a group that contains 5 functions and 3 data symbols.      Its up to the compiler/user to ensure the the elements are the same across compilation units.

Comment (dschuff) [reply Linking.md]: Makes sense.

**Comment by dschuff** [comment]
The new section is pretty short now; is it worth keeping or should it just be folded somewhere else?

**Review by sbc100 — APPROVED** [review]

**Pull request merged**